python/CMakeLists.txt

```diff
@@ -17,3 +17,47 @@ target_compile_definitions(manifpy PRIVATE EIGEN_DEFAULT_TO_ROW_MAJOR)
 set_property(TARGET manifpy PROPERTY CXX_STANDARD 11)
 set_property(TARGET manifpy PROPERTY CXX_STANDARD_REQUIRED ON)
 set_property(TARGET manifpy PROPERTY CXX_EXTENSIONS OFF)
+
+# Detect the active site-packages folder instead of using CMAKE_INSTALL_PREFIX
+option(
+  DETECT_ACTIVE_PYTHON_SITEPACKAGES
+  "Do you want manif to detect and use the active site-package directory? (it could be a system dir)"
+  FALSE)
+
+# Installation in the active site-packages folder (could be a system folder)
+if(DETECT_ACTIVE_PYTHON_SITEPACKAGES)
+  set(PYTHON_INSTDIR "${Python3_SITELIB}/manifpy")
+# Installation from the build extension of setup.py
+elseif(CALL_FROM_SETUP_PY)
+  set(PYTHON_INSTDIR "${CMAKE_INSTALL_PREFIX}")
+# Installation in the CMAKE_INSTALL_PREFIX
+else()
+  execute_process(
+    COMMAND
+    ${Python3_EXECUTABLE} -c
+    "import distutils.sysconfig; print(distutils.sysconfig.get_python_lib(plat_specific=True, standard_lib=False, prefix=''))"
+    OUTPUT_VARIABLE _PYTHON_INSTDIR)
+  string(STRIP ${_PYTHON_INSTDIR} _PYTHON_INSTDIR_CLEAN)
+  set(PYTHON_INSTDIR "${_PYTHON_INSTDIR_CLEAN}/manifpy")
+endif()
```

Comment:
So, I don't quite understand this chunk. Correct me if I'm wrong but the overall idea is to choose whether to install in the CMake install path or the Python's site-package right? 

Now the logic seem odd; aren't the two last blocks inverted? Shouldn't it be 
```cmake
elseif(CALL_FROM_SETUP_PY)
  execute_process(...)
else()
  set(PYTHON_INSTDIR "${CMAKE_INSTALL_PREFIX}")
```
??

Furthermore, the variable `Python3_SITELIB` is provided by the FindPython3 module and is the results of `distutils.sysconfig.get_python_lib(plat_specific=True, standard_lib=False)` (see [doc](https://cmake.org/cmake/help/v3.14/module/FindPython3.html)). So why do we have two distinct cases leading to the same result?

I don't know what's the common practice (if any) but shouldn't we install by default in the Python site-package?

Reply:
Scratch my previous comment about the 'if' block logic being odd, I went down the rabbit hole :sob:.
Still, the `CMAKE_INSTALL_PREFIX` variable populated by `cmake-build-extension` is anything but a prefix :sweat_smile:.

Reply:
Yeah sorry for replying too late, it would have saved you quite some digging :sweat_smile: For the records, in our projects we always find odd when a CMake projects installs files outside `CMAKE_INSTALL_PREFIX`, this is the rationale behind this logic. This is a behavior that could surprise users and third-party packagers and we try to avoid it.

However, in this specific case, using the detected `site-packages` folder (that could be a system folder) makes sense in some use cases. In fact, installing in a custom `CMAKE_INSTALL_PREFIX`, in most cases, requires the user to extend manually `PYTHONPATH` if they are using the system interpreter. Nowadays I believe this is not the most common case and this is the reason why the default behavior of this CMake logic in to install the bindings inside the `CMAKE_INSTALL_PREFIX`, where the right installation path has to be structured. Here is where `distutils` enters in the picture.

Reply:
No worries, it's actually a good thing that I investigated as it allowed me to better understand what's going on.
I generally agree with your point on installing things under the `CMAKE_INSTALL_PREFIX` but for the particular case of python bindings installed from plain CMake, I'm not sure if that makes sense and at the very least it is not convenient since it may requires (as you noted) to augment the `PYTHONPATH`. Now, I don't actually expect anyone to use the plain CMake install for the bindings tho, and I'd encourage using `pip`.